Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1587,4 +1590,235 @@ void Node::build_accessibility_tree(AccessibilityTreeNode& parent) const
+        //   If the embedded control has role textbox, return its value.
+        //   If the embedded control has role menu button, return the text alternative of the button.
+        //   If the embedded control has role combobox or listbox, return the text alternative of the chosen option.
+        //   If the embedded control has role range (e.g., a spinbutton or slider):
+        //   If the aria-valuetext property is present, return its value,
+        //   Otherwise, if the aria-valuenow property is present, return its value,
+        //   Otherwise, use the value as specified by a host language attribute.
+
+        // F. Otherwise, if the current node's role allows name from content, or if the current node is referenced by aria-labelledby, aria-describedby, or is a native host language text alternative element (e.g. label in HTML), or is a descendant of a native host language text alternative element:
+        auto role = element->role_or_default();
+        if (role.has_value() && Web::ARIA::allows_name_from_content(role.value())) {
+            // i. Set the accumulated text to the empty string.
+            total_accumulated_text.clear();
+            // ii. Check for CSS generated textual content associated with the current node and include it in the accumulated text. The CSS :before and :after pseudo elements [CSS2] can provide textual content for elements that have a content model.
+            auto before = element->get_pseudo_element_node(CSS::Selector::PseudoElement::Before);
+            auto after = element->get_pseudo_element_node(CSS::Selector::PseudoElement::After);
+            // For :before pseudo elements, User agents MUST prepend CSS textual content, without a space, to the textual content of the current node.
+            if (before)
+                Node::prepend_without_space(total_accumulated_text, before->computed_values().content().data);
+
+            // For :after pseudo elements, User agents MUST append CSS textual content, without a space, to the textual content of the current node.
+            if (after)
+                Node::append_without_space(total_accumulated_text, after->computed_values().content().data);
+
+            // iii. For each child node of the current node:
+            element->for_each_child([&total_accumulated_text, current_node, target, &document, visited_nodes](DOM::Node const& child_node) mutable {
+                if (visited_nodes.contains(child_node.id()))
+                    return;
+
+                // a. Set the current node to the child node.
+                current_node = &child_node;
+
+                // b. Compute the text alternative of the current node beginning with step 2. Set the result to that text alternative.
+                auto result = MUST(current_node->name_or_description(target, document, visited_nodes));
+
+                // c. Append the result to the accumulated text.
+                total_accumulated_text.append(result);
```

Comment:
This function returns `ErrorOr`, can we just `TRY()` here and use failable methods like `try_append()` instead?

Reply:
Unfortunately, it does not look like we can TRY here because we are inside the for_each_child lambda and there is no version of for_each_child that propagates errors.